### PR TITLE
Python string formatting

### PR DESCRIPTION
**Pull request by hankaj**
This PR changes most of the `%`-formatting in python to f-strings (or `format()` in some cases).
According to Python documentation f-strings and `format()` are preferred to the `%`-formatting (which was used in Python 2).

### CODE REVIEW

**Comment by hankaj** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by hankaj** [comment]
Jenkins CI successfully triggered : [unix-gpu]

**Comment by hankaj** [comment]
@mxnet-bot run ci [unix-gpu]

**Review by anko-intel — CHANGES_REQUESTED** [review]

File: tests/nightly/model_backwards_compatibility_check/common.py [hunk 1]
```diff
@@ -105,10 +105,10 @@ def get_top_level_folders_in_bucket(s3client, bucket_name):
     result = bucket.meta.client.list_objects(Bucket=bucket.name, Delimiter=backslash)
     folder_list = list()
     if 'CommonPrefixes' not in result:
-        logging.error('No trained models found in S3 bucket : %s for this file. '
-                      'Please train the models and run inference again' % bucket_name)
-        raise Exception("No trained models found in S3 bucket : %s for this file. "
-                        "Please train the models and run inference again" % bucket_name)
+        logging.error('No trained models found in S3 bucket : {} for this file. '
```
[the remark this answers is not in the document]
Comment (hankaj) [reply tests/nightly/model_backwards_compatibility_check/common.py:108]: In CI I got the error that you should not use f-strings in messages of logging functions.

File: tests/python/unittest/test_profiler.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def test_profile_create_domain_dept():
 def test_profile_task():
     def makeParams():
         objects = tuple('foo' for _ in range(50))
-        template = ''.join('{%d}' % i for i in range(len(objects)))
+        template = ''.join(f'{{{i}}}' for i in range(len(objects)))
```
[the remark this answers is not in the document]
Comment (hankaj) [reply tests/python/unittest/test_profiler.py:99]:
```
>>> template = ''.join(f'{{{i}}}' for i in range(3))
>>> template
'{0}{1}{2}'
```
This syntax error seems to occur in Python2 and as far as I know it works in Python3. Which python version did you use?

[the remark this answers is not in the document]
Comment (anko-intel) [reply tests/python/unittest/test_profiler.py:99]: You are right. By accident I tested in on python2. It is ok on python3

**Comment by bgawrych** [comment]
@mxnet-bot run ci [sanity]

**Review by bgawrych — APPROVED** [review]
reviewed 2x - LGTM

**Comment by bgawrych** [comment]
@mxnet-bot run ci [all]

**Review by anko-intel — APPROVED** [review]

**Comment by hankaj** [comment]
@mxnet-bot run ci [unix-gpu]

**Pull request merged**